File: Queue/Practice.java
class CircularQueue
{
    public int Arr[];
    public int iFront;
    public int iRare;
    public int iSize;

    CircularQueue(int iSize)
    {
        iFront=-1;
        iRare=-1;
        this.iSize=iSize;
        Arr=new int[iSize];
    }
    public boolean isEmpty()
    {
        if(iFront==-1 && iRare==-1)
        {
            return true;
        }
        else
        {
            return false;
        }
    }
    public boolean isFull()
    {
        if(iFront==0 && iRare==iSize-1 || iRare+1%iSize==iFront)
        {
            return true;
        }
        else
        {
            return false;
        }
    }
    public void enQueue(int iNo)
    {
        if(isFull())
        {
            System.out.println("Queue is empty");
            return;
        }
        if(iFront==iRare)
        {
            iFront=-1;
            iRare=-1;
        }
        else
        {
            
        }
    }
    public void deQueue()
    {

    }


}
class Practice 
{
    public static void main(String agebs[])
    {

    }
}

File: Queue/Practice.cpp
#include<iostream>
using namespace std;
class Queue
{
    public:
    int Front;
    int Rear;
    int *Arr;
    int iSize;

    Queue()
    {
        Front=Rear=-1;
        this->iSize=10;
        Arr=new int[iSize];
    }
    bool empty()
    {
        if(Front==-1 && Rear ==-1)
        {
            return true;
        }
        else
        {
            return false;
        }
    }
    bool full()
    {
        if(Front==0 && Rear == iSize-1 || Rear+1%iSize==Front)
        {
            return true;
        }
        else{
            return false;
        }
    }
    void push(int val)
    {
        if(full())
        {
            cout<<"Queue is full"<<endl;
            return ;
        }
        if(empty())
        {
            Front++;
            Rear++;
            Arr[Rear]=val;
        }
        else
        {
            Rear=((Rear+1)%iSize);
            Arr[Rear]=val;
        }
        
    }
    void pop()
    {
        if(empty())
        {
            cout<<"Queue is empty"<<endl;
            return;
        }
        if(Front==Rear)
        {
            Front=-1;
            Rear=-1;
        }
        else
        {
            Front=(Front+1)%iSize;
        }
    }
    void display()
    {
        int i=0;
        for(i=Front;i!=Rear;i=(i+1)%iSize)
        {
            cout<<i<<" ";
            cout<<Arr[i]<<" ";
        }
        cout<<Arr[i]<<" ";
        cout<<endl;
    }
};
int main()
{
    Queue *obj=new Queue();

    obj->push(10);
    obj->push(10);
    obj->push(10);
    obj->push(10);
    obj->push(10);
    obj->push(10);
    obj->push(10);
    obj->push(10);
    obj->push(10);
    obj->push(10);

    obj->pop();
    obj->pop();
    obj->pop();
    obj->pop();
    obj->push(10);
    obj->push(10);
    obj->push(10);
    obj->push(10);

    

    obj->display();
    return 0;
}